src/report.ts

```diff
@@ -30,27 +34,55 @@ async function pollReport(
     throw new Error(`"project-name" must not contain spaces or special characters except [/-_]`);
   }
 
+  return initAndPollReportGeneric(initReport, getReport, options);
+}
+
+async function initAndPollScmReport(
+  options: ScmUploadReportOptions,
+): Promise<Result<AnalysisFailedResponse | ReportResult, GetAnalysisErrorCodes>> {
+  const projectId = options.projectId?.trim();
+  if (!projectId || projectId.length === 0) {
+    throw new Error('"project-id" must be provided for "report"');
+  }
+  if (!uuid.validate(projectId)) {
```

Comment:
Agreed it can be confusing, but until we can figure out how to align with other product lines and link to platform projects consistently, this is about the best we can do, without adding yet more CLI args (which can also make things more confusing).
More details [here](https://snyk.slack.com/archives/C0370TRPF7F/p1689063822455629?thread_ts=1688130422.138539&cid=C0370TRPF7F).

Reply:
Can't we resolve `project-id` from the `project-name`?

> until we can figure out how to align with other product lines and link to platform projects consistently

As soon as we add it like that, we don't have an easy path back, since this will break customers upon a change.

Reply:
We can only resolve a project name to an ID in our backend currently, after calling the report init API. And that's an idempotent operation that also creates a project with that name if it doesn't exist, so there will be no easy way to do that from the client-side.

However, the path I see going forward to allow users to use `project-id` for both File and SCM flows (and that we discussed with Noa and Andy) is to resolve the project origin type from the ID. That should be a lot easier to do on the client-side, and then we don't need any additional args and can maintain consistency between the commands. I.e. users will be able to run `snyk code test --project-id="<CLI or SCM project ID>" --commit-id="<...>"` and we will detect the type of flow based on the project's origin. If they provide a CLI project ID we do the file flow, otherwise the SCM flow.

Reply:
Yeah, that'd be more consistent. Will you have a work item for that?

Reply:
Added as a point for phase 1.5 in the [brief](https://docs.google.com/document/d/1z_QHy8VbDsccFHydSQSU94gp0QN0b0aRvy6J1EVRs0o/edit).
It's going to be down the line though, but will give us a migration path that won't break existing customers' flows while keeping the CLI args to a minimum, so definitely worth doing.